### ISSUE TITLE
[PAID] [3rd Party Feeds] Implement API calls for 3rd part workspace feeds v1 

### ISSUE THREAD

**Issue by dylanexpensify** [Expensify/App#49690]
Coming from [this thread](https://expensify.slack.com/archives/C06ML6X0W9L/p1727255673116329?thread_ts=1727124415.008809&cid=C06ML6X0W9L), we have some clean up to-dos for the Third-party workspace feeds v1 initiative. 

<details><summary>Issue Owner</summary>Current Issue Owner: @strepanier03</details>

**Comment by VickyStash** [comment]
Hi, I'm Viktoryia from Callstack - expert contributor group - and I would like to work on this issue.

**Comment by robertjchen** [comment]
### WorkspaceCompanyCardsPage.tsx

- **OpenPolicyCompanyCardsPage**(policyID)

### WorkspaceCompanyCardsListHeaderButtons.tsx
- **AssignCard**(policyID, bankName, encryptedCardNumber, email, startDate, reportActionID)

### WorkspaceCompanyCardsSettingsPage.tsx
### WorkspaceCompanyCardFeedSelectorPage.tsx
### WorkspaceMemberNewCardPage.tsx
- **OpenPolicyCompanyCardsFeed**(policyID, bankName)
- **SetCardExportAccount**(authToken, cardID, exportAccountDetails)
- **SetCardName**(authToken, cardID, cardName)

### CardSelectionStep.tsx
- **OpenPolicyCompanyCardsFeed**() called earlier in the flow should already load the list of cards 🤔

**Comment by narefyev91** [comment]
Hi, I'm Nicolay from Callstack - expert contributor group - will help here as well (from tomorrow).
Will pick:
- WorkspaceCompanyCardsSettingsPage.tsx
- WorkspaceMemberNewCardPage.tsx
OpenPolicyCompanyCardsFeed(policyID, bankName)
SetCardExportAccount(cardID, exportAccountDetails)
SetCardName(cardID, cardName)

**Comment by joekaufmanexpensify** [comment]
Assigned!

**Comment by VickyStash** [comment]
@robertjchen I have a question about **reportActionID** for this API call:
**AssignCard**(policyID, bankName, encryptedCardNumber, email, startDate, reportActionID).

Is it for this message? Should I optimistically create it?
![image](https://github.com/user-attachments/assets/71c96bb7-8771-4519-9425-e44b0df8c889)

I'm asking cause it looks like we don't create similar messages optimistically while issuing Expensify Cards.
^^ @mountiny

**Comment by robertjchen** [comment]
Interesting! That's a good point- if these messages are not being optimistically created, then the `reportActionID` is not needed 👍

**Comment by mountiny** [comment]
@robertjchen I think it's a bit different thought, right? With Expensify cards, we have to talk to Marqeta, a third-party service, and so there is a higher chance of the call with fail, so we do not create the actions optimistically. In this case, when we already have the list of the available cards, there are fewer things that can cause it to fail, so I think we could consider adding these optimistically; what do you think, @robertjchen?

**Comment by robertjchen** [comment]
That makes sense, that context definitely helps!

Since we already have all the info we need in Onyx at time of assignment, we can definitely do it optimistically which would make for a better user experience

**Comment by mountiny** [comment]
@VickyStash Feel free to add the optimistic reportActionID param to the api call and create the action optimistically and we can add it to BE

**Comment by robertjchen** [comment]
@VickyStash It would be `CARDASSIGNED`

**Comment by robertjchen** [comment]
Yes, it appears to be under review at the moment 👍

**Comment by dylanexpensify** [comment]
Making progress

**Comment by dylanexpensify** [comment]
TY @VickyStash!

**Comment by VickyStash** [comment]
**Updates:**
- I was finally able to create a feed using **RequestFeedSetup** API call!
- Fixed Feed switcher display in pending state
- Fixed feature check at Update Feed name
- Added some TS/lint fixes
- Reported a couple of API related issues and waiting for test setup of feed with cards

**Comment by VickyStash** [comment]
**Updates:**
- Integrated **OpenPolicyCompanyCardsFeed** API call
- Fixed card feed selector
- Waiting for API fixes to be unblocked

**Comment by VickyStash** [comment]
**Updates:**
- Updated types and data structure to follow BE updates
- Fixed No Cards view
- Fixed card assignment flow (real-life card selection instead of mocked)
- A lot of minor checks/UI fixes

https://github.com/user-attachments/assets/fa194161-8e8a-4ff0-b0a2-733f85417c14

**TODOs (waiting for responses in slack):**
- Card names storing/display (slack [link](https://callstack-hq.slack.com/archives/C06ML6X0W9L/p1728467135670659?thread_ts=1727729604.987049&cid=C06ML6X0W9L))
- Card details data display: `Last updates` and `Transaction start date` (slack [link](https://callstack-hq.slack.com/archives/C06ML6X0W9L/p1728467355721529?thread_ts=1727729604.987049&cid=C06ML6X0W9L))
- CARDASSIGNED report action display - should be fixed by API (slack [link](https://callstack-hq.slack.com/archives/C06ML6X0W9L/p1728469195187489?thread_ts=1727729604.987049&cid=C06ML6X0W9L))
- Update card (sync) - can be implemented in a separate PR (slack [link](https://callstack-hq.slack.com/archives/C06ML6X0W9L/p1728467422517249?thread_ts=1727729604.987049&cid=C06ML6X0W9L))
- Getting data to display all workspace member assigned third-party cards (slack [link](https://callstack-hq.slack.com/archives/C06ML6X0W9L/p1728482728126749?thread_ts=1727729604.987049&cid=C06ML6X0W9L))
- Improve Company Cards screen loading on the first open after login
- Remove Card Feed - can be implemented in a separate PR

**Comment by VickyStash** [comment]
Not a lot of updates from [yesterday](https://github.com/Expensify/App/issues/49690#issuecomment-2402689584):
- Updated member details page to display all of the member cards.
- Waiting for API updates mentioned above

**Comment by VickyStash** [comment]
**Updates:**
- Implemented custom card names storing/display
- Waiting for API fixes

**Comment by VickyStash** [comment]
**Updates:**

- Implemented some improvements related to card details display after the latest API updates got merged.
- Resolved conflicts caused by the latest main, there were pretty much of them since in parallel the work on v2 is going on.
- Fixed minor issues.
- Added test steps and part of the screenshots to the PR.

**TODOs:**
Test the app on mobile platforms and record the videos. After that, I'll open the PR for review. If everything is smooth it should be tomorrow first part of the day.

**Comment by VickyStash** [comment]
The [PR](https://github.com/Expensify/App/pull/49928) has been opened for the review.

**Comment by dylanexpensify** [comment]
Reassigning as I head to parental leave next week! Thanks Sheena!

**Comment by mountiny** [comment]
$250 to @narefyev91

**Comment by mountiny** [comment]
psst

**Comment by mountiny** [comment]
New feature so no checklist needed here

**Comment by strepanier03** [comment]
### Payment Summary

- [x] @VickyStash is from an agency-contributor and not due payment
- [x] @narefyev91 is from an agency-contributor and not due payment
- [x] $250 - @allgandalf paid $250 via Upwork ([LINK](https://www.upwork.com/nx/wm/workroom/38847849/overview?milestone=39261833))

**Comment by strepanier03** [comment]
The contact list is an old process we don't do any longer, not sure why it's resurfaced. https://expensify.slack.com/archives/C01SKUP7QR0/p1729883091032869
